src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedCommand.groovy

```diff
@@ -0,0 +1,63 @@
+package de.dkfz.roddy.execution.jobs.cluster
+
+import de.dkfz.roddy.execution.jobs.BEJob
+import de.dkfz.roddy.execution.jobs.BEJobID
+import de.dkfz.roddy.execution.jobs.BatchEuphoriaJobManager
+import de.dkfz.roddy.execution.jobs.ProcessingParameters
+import de.dkfz.roddy.execution.jobs.SubmissionCommand
+import de.dkfz.roddy.tools.LoggerWrapper
+import groovy.transform.CompileStatic
+
+import java.util.logging.Level
+
+@CompileStatic
+abstract class GridEngineBasedCommand extends SubmissionCommand {
+
+    private static final LoggerWrapper logger = LoggerWrapper.getLogger(GridEngineBasedCommand.class.name)
+    public static final String NONE = "none"
+
+    /**
+     * A command to be executed on the cluster head node, in particular qsub, bsub, qstat, etc.
+     *
+     * @param parentJobManager
+     * @param job
+     * @param jobName
+     * @param processingParameter s@param environmentVariables *@param dependencyIDs @param command
+     */
+    GridEngineBasedCommand(BatchEuphoriaJobManager parentJobManager, BEJob job, String jobName, List<ProcessingParameters> processingParameters, Map<String, String> environmentVariables, List<String> dependencyIDs, String command) {
+        super(parentJobManager, job, jobName, processingParameters, environmentVariables, dependencyIDs, command)
+    }
+
+
+    @Override
+    protected String assembleDependencyString(List<BEJobID> jobIds) {
+        StringBuilder qsubCall = new StringBuilder("")
+        LinkedList<String> tempDependencies =
+                jobIds.findAll {
+                    it.getId() != "" && it.getId() != NONE && it.getId() != "-1"
+                }.collect {
+                    it.getId().split("\\.")[0] // Keep the command line short. PBS accepts the job number for dependencies.
+                } as LinkedList<String>
+        if (tempDependencies.size() > 0) {
+            try {
+                qsubCall <<
+                        getDependsSuperParameter() <<
+                        getDependencyParameterName() <<
+                        getDependencyOptionSeparator() <<
+                        tempDependencies.join(getDependencyIDSeparator())
+            } catch (Exception ex) {
+                logger.log(Level.SEVERE, ex.toString())
```

Comment:
What kind of exception can occur here? Is it really necessary to catch everything? Is that really an appropriate recovery -- or would it be better to let the ex. raise further?

Reply:
I don't know. I removed it. Actually with all the improvements for job ids, this should work fine.